frame/lending/src/mocks_offchain/wrapper.rs

```diff
@@ -0,0 +1,150 @@
+use serde::{Deserialize, Serialize};
+use sp_application_crypto::BoundToRuntimeAppPublic;
+use sp_core::{
+	crypto::{key_types, ByteArray, CryptoType, Dummy},
+	U256,
+};
+pub use sp_core::{sr25519, H256};
+use sp_runtime::{
+	codec::{Decode, Encode, MaxEncodedLen},
+	scale_info::TypeInfo,
+	testing::{TestSignature, UintAuthorityId},
+	traits::{IdentifyAccount, OpaqueKeys, Verify},
+	CryptoTypeId, KeyTypeId,
+};
+use std::{cell::RefCell, ops::Deref};
+
+#[derive(
+	Default,
+	PartialEq,
+	Eq,
+	Clone,
+	Encode,
+	Decode,
+	Debug,
+	Hash,
+	Serialize,
+	Deserialize,
+	PartialOrd,
+	Ord,
+	MaxEncodedLen,
+	TypeInfo,
+)]
+
+// The wrapper was added since AppCrypto is not implemented for UintAuthorityId
+pub struct UintAuthorityIdWrapper(pub UintAuthorityId);
+
+impl From<u64> for UintAuthorityIdWrapper {
+	fn from(id: u64) -> Self {
+		UintAuthorityIdWrapper(UintAuthorityId(id))
+	}
+}
+
+impl From<UintAuthorityIdWrapper> for u64 {
+	fn from(id: UintAuthorityIdWrapper) -> u64 {
+		id.0 .0
+	}
+}
+
+impl UintAuthorityIdWrapper {
+	/// Convert this authority ID into a public key.
+	pub fn to_public_key<T: ByteArray>(&self) -> T {
+		let bytes: [u8; 32] = U256::from(self.0 .0).into();
+		T::from_slice(&bytes).unwrap()
+	}
+}
+
+impl CryptoType for UintAuthorityIdWrapper {
+	type Pair = Dummy;
+}
+
+impl AsRef<[u8]> for UintAuthorityIdWrapper {
+	fn as_ref(&self) -> &[u8] {
+		unsafe {
+			let byte: *const _ = &(self.0 .0 as _);
+			std::slice::from_raw_parts(byte, std::mem::size_of::<u64>())
+		}
```

Comment:
you don't need to use unsafe here: https://doc.rust-lang.org/beta/std/primitive.u64.html#method.to_ne_bytes